modin/pandas/dataframe.py

```diff
@@ -3114,35 +3115,53 @@ def _inflate_light(cls, query_compiler):
         ----------
         query_compiler : BaseQueryCompiler
             Query compiler to use for object re-creation.
+        source_pid : int
+            Determines whether a Modin or pandas object needs to be created.
+            Modin objects are created only on the main process.
 
         Returns
         -------
         DataFrame
             New ``DataFrame`` based on the `query_compiler`.
         """
+        if os.getpid() != source_pid:
+            return query_compiler.to_pandas()
```

Comment:
correct me if I'm wrong, but my understanding is that we use `_inflate_light()` only when we unpickle a `modin.DataFrame` from the plasma storage (as if we wanted to read it from the disk, we would use `_inflate_full()`), so my question is, whether it makes sense to transfer the query compiler to workers and only then call `.to_pandas()`? Isn't calling `.to_pandas()` several times on every worker is more expensive than calling it only once in the main process when serializing? What are the benefits of using `._inflate_light()`?

Reply:
this is the only question that bothers me, otherwise, the PR looks good

Reply:
> Isn't calling .to_pandas() several times on every worker is more expensive than calling it only once in the main process when serializing?

Reply:
> Isn't calling .to_pandas() several times on every worker is more expensive than calling it only once in the main process when serializing?

It may be more expensive, but this method allows calculations to run asynchronously, which mitigates this problem (partially) given that worker processes tend to be under-loaded. On the other hand, the memory consumption on the other hand will be much greater, I believe that you are right and we need to make the call in the main process.